torch/onnx/symbolic_opset12.py

```diff
@@ -13,3 +13,58 @@
 def einsum(g, equation, tensor_list):
     tensors = sym_help._unpack_list(tensor_list)
     return g.op("Einsum", *tensors, equation_s=equation)
+
+def nll_loss(g, self, target, weight, reduction, ignore_index):
+    # none reduction : onnx::Constant[value={0}]
+    # mean reduction : onnx::Constant[value={1}]
+    # sum reduction : onnx::Constant[value={2}]
+    reduction = sym_help._maybe_get_const(reduction, 'i')
+    reduction_vals = ['none', 'mean', 'sum']
+    reduction = reduction_vals[reduction]
+
+    # when ignore_index is not specified, ignore_index == onnx::Constant[value={-100}]
+    if sym_help._maybe_get_const(ignore_index, 'i') != -100:
+        if weight.node().mustBeNone():
+            return g.op("NegativeLogLikelihoodLoss", self, target, reduction_s=reduction)
+        else:
+            return g.op("NegativeLogLikelihoodLoss", self, target, weight, reduction_s=reduction)
+
+    # if ignore_index is specified, compute nllloss with no reduction and apply the reduction afterwards
+    nllloss = g.op("NegativeLogLikelihoodLoss", self, target, reduction_s='none')
```

Comment:
is there also a case where weight is present and ignore_index is specified?

Reply:
yes, this is covered with this line and  the logic line 58.
(compute nllloss with none reduction, then execute the reduction as a subgraph)

Reply:
Yes, but that's only division by sum of weight. There also needs to be element wise multiplication by corresponding weight.

Reply:
the multiplication is line 61 and the division is line 66.
Am I missing something?

Reply:
so I meant that there also needs to be multiplication between nllloss and weight